sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/models/HttpLogOptions.java

```diff
@@ -0,0 +1,334 @@
+     * @return The updated HttpLogOptions object.
+     *
+     * @throws NullPointerException If {@code allowedQueryParamName} is {@code null}.
+     */
+    public HttpLogOptions addAllowedQueryParamName(final String allowedQueryParamName) {
+        this.allowedQueryParamNames.add(allowedQueryParamName);
+        this.getClass().getName();
+
+        return this;
+    }
+
+    /**
+     * Gets flag to allow pretty printing of message bodies.
+     *
+     * @return true if pretty printing of message bodies is allowed.
+     */
+    public boolean isPrettyPrintBody() {
+        return prettyPrintBody;
+    }
+
+    /**
+     * Sets flag to allow pretty printing of message bodies.
+     *
+     * @param prettyPrintBody If true, pretty prints message bodies when logging. If the detailLevel does not include
+     * body logging, this flag does nothing.
+     *
+     * @return The updated HttpLogOptions object.
+     */
+    public HttpLogOptions setPrettyPrintBody(boolean prettyPrintBody) {
+        this.prettyPrintBody = prettyPrintBody;
+
+        return this;
+    }
+
+    /**
+     * Gets the {@link HttpRequestLogger} that will be used to log HTTP requests.
+     *
+     * <p>A default {@link HttpRequestLogger} will be used if one isn't supplied.
+     *
+     * @return The {@link HttpRequestLogger} that will be used to log HTTP requests.
+     */
+    public HttpRequestLogger getRequestLogger() {
+        return requestLogger;
+    }
+
+    /**
+     * Sets the {@link HttpRequestLogger} that will be used to log HTTP requests.
+     *
+     * <p>A default {@link HttpRequestLogger} will be used if one isn't supplied.
+     *
+     * @param requestLogger The {@link HttpRequestLogger} that will be used to log HTTP requests.
+     *
+     * @return The updated HttpLogOptions object.
+     */
+    public HttpLogOptions setRequestLogger(HttpRequestLogger requestLogger) {
+        this.requestLogger = requestLogger;
+
+        return this;
+    }
+
+    /**
+     * Gets the {@link HttpResponseLogger} that will be used to log HTTP responses.
+     *
+     * <p>A default {@link HttpResponseLogger} will be used if one isn't supplied.
+     *
+     * @return The {@link HttpResponseLogger} that will be used to log HTTP responses.
+     */
+    public HttpResponseLogger getResponseLogger() {
+        return responseLogger;
+    }
+
+    /**
+     * Sets the {@link HttpResponseLogger} that will be used to log HTTP responses.
+     *
+     * <p>A default {@link HttpResponseLogger} will be used if one isn't supplied.
+     *
+     * @param responseLogger The {@link HttpResponseLogger} that will be used to log HTTP responses.
+     *
+     * @return The updated HttpLogOptions object.
+     */
+    public HttpLogOptions setResponseLogger(HttpResponseLogger responseLogger) {
+        this.responseLogger = responseLogger;
+
+        return this;
+    }
+
+    /**
+     * The level of detail to log on HTTP messages.
+     */
+    public enum HttpLogDetailLevel {
```

Comment:
From an offline discussion, we should look into removing this completely and simplifying logging logic to be an all or nothing type scenario.

Reply:
Follow up - https://github.com/Azure/azure-sdk-for-java/issues/37583